pkg/reconciler/ingress/resources/secret_ce.go

```diff
@@ -33,7 +33,38 @@ func CalculateCertificateHash(secret *corev1.Secret) (string, error) {
 
 // MakeMirrorSecret creates a Secret object that mirrors a TLS Secret
 func MakeMirrorSecret(originSecret *corev1.Secret, certificateHash string) *corev1.Secret {
-	return makeSecret(originSecret, certificateHash, config.IstioNamespace, map[string]string{
+	secret := correctSecretFormat(originSecret)
+	return makeSecret(secret, certificateHash, config.IstioNamespace, map[string]string{
 		labelKeyDomainMappingSecret: labelValueDomainMappingSecret,
 	}, map[string]string{})
 }
+
+func decodeAndEncodePEMBlocks(data []byte) []byte {
+	var encodedData []byte
+	for len(data) > 0 {
+		block, rest := pem.Decode(data)
+		if block != nil {
+			encodedBlock := pem.EncodeToMemory(block)
+			encodedData = append(encodedData, encodedBlock...)
+		}
+		data = rest
+	}
+	if len(encodedData) > 0 && encodedData[len(encodedData)-1] == '\n' {
+		return encodedData[:len(encodedData)-1]
+	}
```

Comment:
Is it really required to remove the trailing newline? I would say that only getting rid of the empty lines withing the Base64 encoded block is actually touching on the pain point we have. Reason I wonder is that I would like to avoid that we actually change **all** secrets here. The majority will have no empty lines in the inner block and we would still change all of them when they have a trailing newline. What do you think?

Reply:
Done, I have also modified the tests to accomodate the same